single-line-diagram-core/src/main/java/com/powsybl/sld/layout/VerticalSubstationLayout.java

```diff
@@ -117,8 +117,11 @@ private double getHeightHorizontalSnakeLines(String vlGraphId, BusCell.Direction
             infosNbSnakeLinesH.getNbSnakeLinesTopBottom().put(BusCell.Direction.TOP, currentNbTop);
             infosNbSnakeLinesH.getNbSnakeLinesVerticalBetween().put(graphId, currentNbLeft);
 
+            double yMin = getGraph().getVoltageLevels().stream().mapToDouble(g -> g.getY()).min().orElse(0.0);
+            double yMax = getGraph().getVoltageLevels().stream().mapToDouble(g -> g.getY() + g.getInnerHeight(layoutParam)).max().orElse(0.0);
+
             // Calculate the snakeline as an horizontal layout
-            polyline = calculatePolylineSnakeLineForHorizontalLayout(layoutParam, node1, node2, increment, infosNbSnakeLinesH);
+            polyline = calculatePolylineSnakeLineForHorizontalLayout(layoutParam, node1, node2, increment, infosNbSnakeLinesH, yMin, yMax);
```

Comment:
Here we are in a substation vertical layout, but within a voltage level (snake lines with two ends in the same voltage level). So you need to calculate yMin and yMax by considering only the current voltage level. Indeed the other voltage levels will be above / below.